### PR TITLE
Add teeracle 

### PR DESCRIPTION
**Pull request by echevrier**
- add teeracle demo
- add teeracle SDK
I built the doc in the same way as the other SDK doc: sidechain and off-chain-worker and have added some of the content of the existing version  https://integritee-network.gitbook.io/integritee-wiki/for-developers/teeracle

I aslo added a small tutorial according https://github.com/integritee-network/teeracle/issues/29

### CODE REVIEW

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md [hunk 1]
```diff
@@ -1,3 +1,110 @@
...
+### **Generic TEEracle**
+Coming soon !
+
+### **Exchange rate oracle**
+We already have a running TEEracle connected to the Integritee parachain that supplies the TEER-USD exchange rate. This exchange rate will be used to offer stable USD fees.  
+The exchange rate values are from two different sources, https://www.coingecko.com/ and https://coinmarketcap.com/api/, and updated once per day.
+
+### **How to use the SDK**
+1. Experiment with the template, running the [teeracle demo](../4.6-demos/4.6.3-demo-demo.md)​
+2. Fork the worker repository `https://github.com/integritee-network/worker.git` from the SDK release branch (TODO release)
```
Comment (echevrier) [inline 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:13]: We have no teeracle release branch for now

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md [hunk 2]
```diff
@@ -1,3 +1,110 @@
...
+3. Build the worker in [teeracle mode](4.4.3-teeracle-oracle-framework.md#build-teeracle-mode)
+4. Customize [ your own exchange rate oracle](4.4.3-teeracle-oracle-framework.md#customize-exchange-rate)
+5. Deploy on the [Integritee parachain](4.4.7-integritee-parachain-integration.md) or your own testnet
```
Comment (echevrier) [inline 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:16]: Shall we add a doc for launching a own testnet? Where?

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md [hunk 3]
```diff
@@ -1,3 +1,110 @@
...
+### **Add trusted source to whitelist**<a href="#add-to-whitelist" id="add-to-whitelist"></a>
+The [teeracle pallet](https://github.com/integritee-network/pallets/tree/master/teeracle/src) maintains a whitelist of trusted exchange rate oracles. A trusted source can be added to the whitelist by Sudo or over the [Democracy](2.5.4-democracy.md)
+
+Call : `teeracle, addToWhitelist ( dataSource , 0x mrenclave_hex)`
+
+The dataSource is the base URL of the source:  https://pro-api.coinmarketcap.com/ or https://api.coingecko.com/. Warning: no spaces and / at the end.   
+The mrenclave hex is written to the logs, when the teeracle service starts.
+
+### **How to build a USD-DOT Oracle from the Template on your own testnet**<a href="#usd-dot-oracle" id="usd-dot-oracle">
+1. Fork the worker repository `https://github.com/integritee-network/worker.git` from the SDK release branch (TODO release)
```
Comment (echevrier) [inline 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:91]: I'm not sure that this tutorial is needed. But if it is, I can improve it.

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md [hunk 4]
```diff
@@ -1,3 +1,110 @@
...
+6. Add the trusted sources to the [Teeracle whitelist](4.4.3-teeracle-oracle-framework.md#add-to-whitelist)
+7. Try the tutorial [How to build a USD-DOT Oracle from the Template on your own testnet](4.4.3-teeracle-oracle-framework.md#usd-dot-oracle)
+
+### **Building the worker in teeracle mode** <a href="#build-teeracle-mode" id="build-teeracle-mode"></a>
+
+In order to build the worker in teeracle mode, the corresponding cargo feature `teeracle` needs to be set. In the Makefiles, the environment variable `WORKER_MODE` is used to set the cargo features.
+
+In case you build with `make` directly, do so with:
+
+`1WORKER_MODE=teeracle make`
```
Comment (brenzi) [inline 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:26]: typo

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md [hunk 5]
```diff
@@ -1,3 +1,110 @@
...
+
+In case you use docker with our `build.Dockerfile`, use `--build-arg WORKER_MODE_ARG=teeracle` to set the corresponding docker `ARG`.
+
+An example of a docker build command (as currently used for GitHub CI):
+
+<pre data-line-numbers><code><strong>docker build -t integritee-worker --target deployed-worker --build-arg WORKER_MODE_ARG=teeracle -f build.Dockerfile .</strong></code></pre>
```
Comment (brenzi) [inline 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:32]: I'd suggest to use markdown for code formatting, not html

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md (hunk 3 above)
Comment (brenzi) [reply 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:91]: I think it's good to have an example, but we can also supply that later if you think it needs reworking

Comment (brenzi) [reply 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:91]: showing this completely local demo setup is valuable for sure

**Review by brenzi — APPROVED** [review]
I like it and I think we can go with this for the release. I'll approve although the release branch needs still to be defined and one typo needs fixing

File: 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md (hunk 3 above)
Comment (echevrier) [reply 4-development/4.4-sdk/4.4.3-teeracle-oracle-framework.md:91]: > I think it's good to have an example, but we can also supply that later if you think it needs reworking

It is a simple example, listing all important steps.

Questions:
- Shall the commands be for dev mode only?
- Should we assume that the user can use the remote attestation ?
- Is there a doc elsewhere on how to run a solo node or shoul I specify it here?

**Pull request merged**